src/main-client.js

```diff
@@ -117,4 +117,95 @@ function getPastBookingsOnDom(bookings, rooms, todaysDate) {
     }
   }
 
-  
\ No newline at end of file
+  function filterRoomsAvailableByDate(jqueryEvent) {
+    event.preventDefault();
+    let availableRooms = [];
+    let date = $('.booking-input-field').val().split('-').join('/')
+    let bookedRooms = jqueryEvent.data.bookings.filter(booking => booking.date === date)
+      jqueryEvent.data.rooms.forEach(room => {
+        let booked = bookedRooms.find(bookedroom => bookedroom.roomNumber === room.number)
+        if(booked === undefined) {
+          availableRooms.push(room);
+        }
+      })
+    if(availableRooms.length === 0 ) {
+      $('.room-booking-container').css('display', 'none')
+      $('.fierce-apology-container').css('display', 'flex')
+      $('.fierce-apology-container').append(
+      `<section class="fierce-apology-paragraph-section">
+            <p class="fierce-apology-paragraph">I'm sorry to inform you that The Cherry Blossom Hotel is overbooked on
+              ${date} and your selected room is no longer available. We regret the error during such a busy 
+              travel time. Please accept our sincerest apologies for the inconvenience.</p>
+          </section>`)
```

Comment:
Could maybe be split out into HTML generation functions which could then be on a job object, and might help split this into a more modular SRP route if you went that way, but for consideration.

Reply:
Just for my understanding, you mean creating a function that just returns the html and pass in the data through params and just call that function here instead of directly placing html?

Reply:
Yup, exactly, could help break out this code a bit if you were looking to segmented it.